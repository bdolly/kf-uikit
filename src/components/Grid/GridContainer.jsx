import React from 'react';
import PropTypes from 'prop-types';
import classes from 'classnames';

/**
 *  Fluidly responsive 960px, 12 cell grid system container based on the css grid spec.
 */
const GridContainer = ({ className, children, collapsed, centered, fullWidth }) => {
  const GridContainerClass = classes(
    'grid-container',
    { 'mx-auto': centered },
    {
      [`grid-container--collapsed${
        collapsed === 'cells' || collapsed === 'rows' ? `-${collapsed}` : ''
      }`]: collapsed !== false,
    },
    { 'grid-container--fullWidth': fullWidth },
    className,
  );

  return <section className={GridContainerClass}>{children}</section>;
};

GridContainer.propTypes = {
  /** Any additional classes to be applied to the gridContainer */
  className: PropTypes.string,
  /** Children elements. */
  children: PropTypes.node,
  /** Collapse margins and gutters. */
  collapsed: PropTypes.oneOf([true, false, 'cells', 'rows']),
  /** wheter or not to apply margin: 0 auto; */
  centered: PropTypes.bool,
  /** wheter or not to apply max-width: 100%; */
  fullWidth: PropTypes.bool,
};

GridContainer.defaultProps = {
  className: '',
  children: null,
  collapsed: false,
  centered: true,
  fullWidth: false,
};


export default GridContainer;
