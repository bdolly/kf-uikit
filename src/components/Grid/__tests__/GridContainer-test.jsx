import React from 'react';
import renderer from 'react-test-renderer';
import GridContainer from '../GridContainer'

[true, 'rows','cells'].map((collapsed)=>{
  [true, false].map((centered)=>{
    [true, false].map((fullWidth)=>{
    

      it(`GridContainer collapsed:${collapsed} ${centered? 'centered': ''} ${fullWidth? 'fullWidth': ''} renders correctly`,()=>{
        const tree = renderer.create(
          <GridContainer {...{collapsed, centered, fullWidth}}>
            <div>Grid Container content</div>
          </GridContainer>
         )
        expect(tree).toMatchSnapshot();
      })


    })
  })
})