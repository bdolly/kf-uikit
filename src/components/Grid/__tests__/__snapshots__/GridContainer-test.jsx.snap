// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer collapsed:cells   renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed-cells"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:cells  fullWidth renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed-cells grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:cells centered  renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed-cells"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:cells centered fullWidth renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed-cells grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:rows   renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed-rows"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:rows  fullWidth renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed-rows grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:rows centered  renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed-rows"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:rows centered fullWidth renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed-rows grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:true   renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:true  fullWidth renders correctly 1`] = `
<section
  className="grid-container grid-container--collapsed grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:true centered  renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed"
>
  <div>
    Grid Container content
  </div>
</section>
`;

exports[`GridContainer collapsed:true centered fullWidth renders correctly 1`] = `
<section
  className="grid-container mx-auto grid-container--collapsed grid-container--fullWidth"
>
  <div>
    Grid Container content
  </div>
</section>
`;
